src/GraphQL/Resolver/ContentThumbnailResolver.php

```diff
@@ -6,88 +6,41 @@
  */
 namespace EzSystems\EzPlatformGraphQL\GraphQL\Resolver;
 
-use Exception;
 use eZ\Publish\API\Repository\Values\Content\Content;
-use eZ\Publish\API\Repository\Values\Content\Field;
-use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
-use eZ\Publish\Core\FieldType;
-use eZ\Publish\SPI\Variation\VariationHandler;
-use EzSystems\EzPlatformGraphQL\GraphQL\DataLoader\ContentLoader;
+use eZ\Publish\SPI\Repository\Strategy\ContentThumbnail\ThumbnailStrategy;
 
 final class ContentThumbnailResolver
 {
-    const THUMBNAIL_VARIATION_IDENTIFIER = 'medium';
-
-    /**
-     * @var \EzSystems\EzPlatformGraphQL\GraphQL\DataLoader\ContentLoader
-     */
-    private $contentLoader;
-    /**
-     * @var \eZ\Publish\Core\FieldType\ImageAsset\AssetMapper
-     */
-    private $assetMapper;
-    /**
-     * @var \eZ\Publish\SPI\Variation\VariationHandler
-     */
-    private $variationHandler;
-    /**
-     * @var \eZ\Publish\Core\FieldType\Image\Type
-     */
-    private $imageFieldType;
+    /** @var \eZ\Publish\SPI\Repository\Strategy\ContentThumbnail\ThumbnailStrategy */
+    private $thumbnailStrategy;
 
     public function __construct(
-        FieldType\Image\Type $imageFieldType,
-        ContentLoader $contentLoader,
-        FieldType\ImageAsset\AssetMapper $assetMapper,
-        VariationHandler $variationHandler)
-    {
-        $this->contentLoader = $contentLoader;
-        $this->assetMapper = $assetMapper;
-        $this->variationHandler = $variationHandler;
-        $this->imageFieldType = $imageFieldType;
+        ThumbnailStrategy $thumbnailStrategy
+    ) {
+        $this->thumbnailStrategy = $thumbnailStrategy;
     }
 
     /**
      * @return array|null array with the thumbnail info, or null if no thumbnail could be obtained for that image
      */
     public function resolveContentThumbnail(Content $content): ?array
     {
-        try {
-            $imageField = $this->getThumbnailImageField($content);
-        } catch (Exception $e) {
-            return null;
-        }
+        $thumbnail = $this->thumbnailStrategy->getThumbnail(
+            $content->getContentType(),
+            $content->getFields(),
+            $content->getVersionInfo()
+        );
```

Comment:
hmmm I've got feeling that `try ... catch` statement should stay here (with proper error logging)

Reply:
From what I see we are not catchin it inside `\eZ\Publish\Core\Repository\Mapper\ContentDomainMapper::buildContentDomainObject` and we did not had problems with it till now as exceptions are catched (without login tho) inside specific strategies.

Reply:
OK